### PR TITLE
[MM-56005] Convert `./components/admin_console/permission_schemes_settings/permission_team_scheme_settings/team_in_list/team_in_list.tsx` from Class Component to Function Component

### PR DESCRIPTION
**Pull request by Syed-Ali-Abbas-Zaidi**
#### Summary

Convert `./components/admin_console/permission_schemes_settings/permission_team_scheme_settings/permission_team_scheme_settings/team_in_list/team_in_list.tsx` from Class Component to Function Component

#### Ticket Link

Fixes https://github.com/mattermost/mattermost/issues/25597
Jira https://mattermost.atlassian.net/browse/MM-56005

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: webapp/channels/src/components/admin_console/permission_schemes_settings/permission_team_scheme_settings/team_in_list/team_in_list.tsx [hunk 1]
```diff
@@ -16,41 +16,39 @@ type Props = {
     isDisabled: boolean;
 }
 
-export default class TeamInList extends React.PureComponent<Props> {
-    handleRemoveTeam = () => {
-        const {team, isDisabled, onRemoveTeam} = this.props;
+const TeamInList = ({team, isDisabled, onRemoveTeam}: Props) => {
```
Comment (larkox) [inline webapp/channels/src/components/admin_console/permission_schemes_settings/permission_team_scheme_settings/team_in_list/team_in_list.tsx:19]: Can you put each prop in a different line?
```suggestion
const TeamInList = ({
    team,
    isDisabled,
    onRemoveTeam,
}: Props) => {
```

File: webapp/channels/src/components/admin_console/permission_schemes_settings/permission_team_scheme_settings/team_in_list/team_in_list.tsx [hunk 2]
```diff
@@ -16,41 +16,39 @@ type Props = {
...
-                        id='admin.permissions.teamScheme.removeTeam'
-                        defaultMessage='Remove'
-                    />
-                </a>
             </div>
-        );
-    }
-}
+            <a
+                className={isDisabled ? 'remove disabled' : 'remove'}
```
Comment (larkox) [inline webapp/channels/src/components/admin_console/permission_schemes_settings/permission_team_scheme_settings/team_in_list/team_in_list.tsx:42]: Can we change this to use `classnames`? (you can see how it is used in other files in the codebase)

If I remember correctly, it should be something like `classnames('remove', {disabled: isDisabled})`.

**Review by larkox — CHANGES_REQUESTED** [review]
Great work! Just 2 minor things.

**Review by larkox — APPROVED** [review]

**Review by hmhealey — APPROVED** [review]
Thanks for the PR. This looks good to me!

**Pull request merged**